proposals/Level-1.md

```diff
@@ -78,94 +77,69 @@ instruction. Thrown exceptions are handled as follows:
 1. Throws not caught within a function body continue up the call stack, popping
    call frames, until an enclosing try block is found.
    
-1. If the call stack is exhausted without any enclosing try blocks, embedder
+1. If the call stack is exhausted without any enclosing try blocks, the embedder
    defines how to handle the uncaught exception.
    
 ### Event handling
 
 This proposal focusses on adding exception handling, but tries to be more
-general in its specification. This allows future generalizations to WebAssembly
-without having to redo exception handling.
+general in its specification such that more general kinds of event handling can
+be added to WebAssembly in the future without having to redo exception handling.
 
-Events are generated by event yielding instructions. They yeild to an
-appropriate event handler. The event handler is then run. That event handler
-can either resume control at the point of the event handler, or optionally
-resume by sending values back to the yielding instruction allow the
-originating code to resume.
+An event handler allows one to process an event generated by a block of
+code. Events yield the current execution and look for a corresponding event
+handler. If found, the corresponding event handler is run.  The event handler
+can optionally resume by sending values back to the yielded instruction,
+allowing the originating code to resume.
 
 Exceptions are a special case of an event in that they never resume. Similarly,
-a throw instruction is the yielding instruction of an exception, and the
-corresponding catching block is the event handler of the exception.
+a throw instruction is the yielding event of an exception. The catch block
+associated with a try block defines how to handle the throw.
```

Comment:
Is this implying that other kinds of events will have different instructions that go with them?